### ISSUE TITLE
DataGrid Accessibility - Can't Tab to Actions Column

### ISSUE THREAD

**Issue by md250721** [mui/mui-x#9250]
### Duplicates

- [X] I have searched the existing issues

### Latest version

- [X] I have tested the latest version

### Steps to reproduce 🕹

Link to live example: https://codesandbox.io/s/ccqysc?file=/demo.tsx

Steps:
1. Press the edit icon on a row's actions column
2. Begin tabbing
3. Notice that you cannot tab all the way to the Save icon and that the tabbing stops at the "Last Login" column

### Current behavior 😯

You cannot tab all the way to the Save icon and that the tabbing stops at the "Last Login" column

### Expected behavior 🤔

You should be able to continue tabbing all the way to the Save action and be able to save your edits with just the keyboard.

### Context 🔦

The main context is making sure that our grid is accessible to keyboard users.

### Your environment 🌎

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
I used Chrome 

System:
    OS: macOS 13.3.1
  Binaries:
    Node: 19.5.0 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 9.3.1 - /usr/local/bin/npm
  Browsers:
    Chrome: 114.0.5735.106
    Edge: 114.0.1823.37
    Firefox: Not Found
    Safari: 16.4
  npmPackages:
    @emotion/react:  11.10.6 
    @emotion/styled:  11.10.6 
    @mui/base: ^5.0.0-alpha.92 => 5.0.0-alpha.126 
    @mui/core-downloads-tracker:  5.12.1 
    @mui/icons-material: ^5.6.2 => 5.8.0 
    @mui/material: ^5.12.1 => 5.12.1 
    @mui/private-theming:  5.12.0 
    @mui/styled-engine:  5.12.0 
    @mui/system: ^5.9.1 => 5.12.1 
    @mui/types:  7.2.4 
    @mui/utils:  5.12.3 
    @mui/x-data-grid:  5.12.3 
    @mui/x-data-grid-pro: ^5.12.3 => 5.12.3 
    @mui/x-date-pickers:  5.0.20 
    @mui/x-date-pickers-pro: ^5.0.15 => 5.0.15 
    @mui/x-license-pro:  5.12.3 
    @types/react: ^18.0.8 => 18.0.9 
    react: ^18.1.0 => 18.2.0 
    react-dom: ^18.1.0 => 18.1.0 
    styled-components:  5.3.8 
    typescript: ^4.6.4 => 4.7.2 
```
</details>

**Comment by romgrk** [comment]
We should probably review keyboard access in the rest of the grid as well. IIRC the header filters weren't really usable KB-only, and the headers' menu can't be toggle with the keyboard.

**Comment by MBilalShafi** [comment]
> You should be able to continue tabbing all the way to the Save action and be able to save your edits with just the keyboard.

It's a valid use case I guess, though as a rule of thumb, Tab only works for the cells that are actually editable while the row is in edit mode, other than that we follow arrow keys navigation everywhere else as explained in the [WCAG accessibility pattern](https://www.w3.org/WAI/ARIA/apg/patterns/grid/)

> the header filters weren't really usable KB-only, and the headers' menu can't be toggled with the keyboard.

Can you mention a specific example, I guess you can do all the actions on both headers and header filters using keyboard, to open menu on both a header filter and header cell, you can press <kbd>Ctrl/Cmd</kbd> + <kbd>Enter</kbd>, it's documented [h](https://mui.com/x/react-data-grid/column-menu/)er[e](https://mui.com/x/react-data-grid/filtering/header-filters/).

**Comment by md250721** [comment]
@romgrk @MBilalShafi - do you have a general timeline of when the `Tab` issue could be resolved? thanks

**Comment by md250721** [comment]
@m4theushw thanks! Yes I can submit a PR for this tomorrow

**Comment by md250721** [comment]
@m4theushw - here is the PR: https://github.com/mui/mui-x/pull/9321
Please let me know if there are any other changes needed
